test/api.test.js

```diff
@@ -454,6 +454,37 @@ test('findOrCreate', (t) => {
   })
 })
 
+test('findOrCreate (metadata does not collide)', (t) => {
+  const sbot = Testbot()
+
+  const details = {
+    feedpurpose: 'chess',
+    // feedformat: 'classic', optional
+    metadata: {
+      type: 'dogs',
+    },
+  }
+
```

Comment:
here's the test. Without the changes to the code the latest branch you get from the stream is the "shard" (as the message published that was type `metafeed/add/derived` gets replaced by type `dogs` :dog:  Woof

Reply:
I would rather that this gives an error. You are not supposed to use any reserved words in metadata.